### PR TITLE
FAVCS-99: "Apply date" missing from the jobfeed

### PR DESCRIPTION
**Pull request by mnakov**
### Changes
- Translate date months from Danish to English
  
### Technical Details
- Add custom tamper to convert dates from Danish to English
- Add tamper to 'deadline' field and export feature
  
### Affected Pages
- /job
  
### Key Affected Code
- docroot/profiles/favrskov/modules/custom/favrskov_helper/feeds_tamper_plugins/date_converter.inc
  
### Key Functionality in custom code
- N/A
  
### References
- N/A
  
### Notices
- After deploye run: `drush fr feeds_job_importer`

### CODE REVIEW

**Comment by mnakov** [comment]
@vzlatev, the old code is not relative anymore as the date format from the source is different. It comes in this format "29. maj 2019" which is not timestamp so we should convert it manually. Here is an error which is triggered by the old date convertor tamper logic: `DateTime::__construct(): Failed to parse time string (29. maj 2019) at position 0 (2): Unexpected character.` So Miroslav implemented the months translation logic and I just moved it to the old tamper and modified it.

**Comment by vzlatev** [comment]
@mnakov 

I understand the situation. My last questions is have you tried to use https://www.php.net/manual/en/function.setlocale.php as this can be much cleaner solution with fewer lines of code? In theory when you set proper locale (english one) maj will automatically become May.

**Comment by vzlatev** [comment]
@mnakov 
Please check Jira ticket and adapt the code.

File: docroot/profiles/favrskov/modules/custom/favrskov_helper/feeds_tamper_plugins/date_converter.inc [hunk 1]
```diff
@@ -36,8 +36,13 @@ function favrskov_helper_plugin_date_to_proper_format_form($importer, $element_k
  * Callback that will be called during import on field that plugin was attached.
  */
 function favrskov_helper_plugin_date_to_proper_format_callback($result, $item_key, $element_key, &$field, $settings) {
-  $date       = new DateTime($field);
-  $timestamp  = $date->getTimestamp();
 
-  $field = format_date($timestamp, $settings['type']);
+  $current_local = locale_get_default();
+  setlocale(LC_ALL,"da_DK");
+  $date = DateTime::createFromFormat("d. M Y", $field);
```
Comment (vzlatev) [inline docroot/profiles/favrskov/modules/custom/favrskov_helper/feeds_tamper_plugins/date_converter.inc]: @mnakov Are you sure about the dot as you mentioned that some days may not have it ?

Comment (mnakov) [reply docroot/profiles/favrskov/modules/custom/favrskov_helper/feeds_tamper_plugins/date_converter.inc]: @vzlatev, I'm sure about the dot, the date always comes with the dot. You can proceed with merge, if everything else seems good to you.

**Pull request merged**